### ISSUE TITLE
UI Styling + Accessibility enhancement

### ISSUE THREAD

**Issue by xNS5** [xNS5/rentalreviews#20]
Clean up the UI to look nicer, taking accessibility into account. I wanted this to be a separate ticket so I could focus on base functionality for the site.

**Comment by xNS5** [comment]
Pages: 

- [x] Home
- [x] Reviews
- [x] Review
- [x] Data
- [x] FAQ
- [x] Privacy Policy
- [x] Navbar
- [x] Mobile Navbar

**Comment by xNS5** [comment]
The Home page and the navbar + footer are done to my liking. I switched out the hover effect to a simple underline, I didn't really care for the blue background + white text color effect on all Link components.

**Comment by xNS5** [comment]
To go into more detail about the hover effect, I decided to remove the dedicated "Read" button in favor of just making the property/company name the hyperlink. I would have had to add that same effect onFocus/FocusOut (or whatever the function name is) and that sounded like a pain for a simple visual effect.

Regarding accessibility of the data table, I decided to remove the `tabindex=0` on the row in favor of adding `tabindex=0` to the cell components instead. The next trick was figuring out how to get the aria-label to say anything meaningful, so I opted to create a `alt` config in the database which will store alt text for situations like this. Why a database call? It seemed more in line with the other configuration files, and as they're globally applicable I thought it would be a fitting place. 

Will start making some UI tweaks in the AM.

**Comment by xNS5** [comment]
Note to self: in the `review.tsx` page, the review data should have the same aria-labels as the data table. For the tooltips, I'm thinking maybe just to set the value of the tooltip to the aria-label or something.

**Comment by xNS5** [comment]
Running into some issues with the data table. If I have the sort buttons enabled, then I can't use the built-in screen reader effectively. If they're disabled, I can use the keyboard to navigate without an issue. 

This brings me to another question: how can I add functionality that supports users with screen readers and those who don't use one? 

For the navbar, I'm getting some console errors relating to the <Image/> component I threw together, might want to consider switching to some kind of accordion/hamburger menu instead of the modal type menu.

**Comment by xNS5** [comment]
Found out what was going wrong with the data table, and it was user error on my part. I was under the impression that the screen reader required tab navigation to go from element to element, however after doing some research I found out that I was supposed to use the "T" key with the screen reader, and then use arrow keys to navigate. 

Now that I have the data table working, and I'm applying aria-labels where necessary. Might want to add something for the `company_type` column, but it's no big deal.

**Comment by xNS5** [comment]
I realized that I'm going to need to test this out on mobile as well. This might require migrating this to a docker container. Ugh.

**Comment by xNS5** [comment]
Finished the data table. The sorting functionality has some...quirks, but I'm going to be adding some more advanced filtering after the MVP release so this is something that can be dealt with later.

**Comment by xNS5** [comment]
I'll be working on the FAQ next, it'll be a bit quicker and I need a break from the data table lol. I'm thinking that I'll nix the accordion components in the FAQ. When I went over it with the screen reader it felt annoying to have to navigate to the question, hear that it's a collapsible button, then expand it to hear the answer. That's, what, one extra keyboard input than necessary?

**Comment by xNS5** [comment]
After doing some testing on `/data`, I've concluded that the component I was previously using has some accessibility issues which makes keyboard navigation a challenge. I tried another popular JSON viewer component, however I'm finding that while it's somewhat more accessible, there is much to be improved. For example, `onMouseEnter()` renders a copy button -- which can be disabled, but the button is a `<span/>` instead of an actual `<button/>` component. And then, there's some logic that collapses strings after a certain length -- which again isn't ideal for screen readers. 

For the time being, I'm going to formally merge `rest-api` branch with 27 so I can utilize the browser's built-in JSON viewer. I'm happier with the browser's json viewer in terms of accessibility. It's a clean approach, however I would ultimately prefer to add something in a page by itself. Ah well.
